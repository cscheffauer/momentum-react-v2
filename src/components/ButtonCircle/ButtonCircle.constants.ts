const CLASS_PREFIX = 'md-button-circle';

const DEFAULTS = {
  COLOR: 'primary',
  DISABLED: false,
  SHALLOW_DISABLED: false,
  GHOST: false,
  OUTLINE: false,
  SIZE: 40,
  INVERTED: false,
} as const;

const COLORS = {
  JOIN: 'join',
  CANCEL: 'cancel',
  MESSAGE: 'message',
};

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
  widthOverride: `${CLASS_PREFIX}-width-override`,
};

export { CLASS_PREFIX, COLORS, DEFAULTS, STYLE };
