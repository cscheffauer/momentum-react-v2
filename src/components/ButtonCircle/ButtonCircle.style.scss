.md-button-circle-width-override {
  padding: unset;
  aspect-ratio: 1 / 1;
}

// for ButtonCircle in ButtonGroup case (to be removed once ButtonGroup is refactored)
[data-compressed='true'] {
  > .md-button-circle-wrapper {
    &[size='64'] {
      width: 3.2rem;

      --mdc-button-prefix-icon-size: 1.5rem;
    }

    &[size='52'] {
      width: 2.6rem;

      --mdc-button-prefix-icon-size: 1.25rem;
    }

    &[size='40'] {
      width: 1.875rem;

      --mdc-button-prefix-icon-size: 1rem;
    }

    &[size='32'] {
      width: 1.5rem;

      --mdc-button-prefix-icon-size: 1rem;
    }

    &[size='28'] {
      width: 1.31rem;

      --mdc-button-prefix-icon-size: 0.75rem;
    }

    // this size is ONLY for AddReactionButton
    &[size='20'] {
      width: 1.2rem;

      --mdc-button-prefix-icon-size: 0.75rem;
    }
  }
}
