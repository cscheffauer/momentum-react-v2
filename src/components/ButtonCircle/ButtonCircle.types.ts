import type { MdcButtonProps } from '../../types';
import { ReactNode } from 'react';

export type Props = Omit<MdcButtonProps, 'active' | 'color'> & {
  onPress?: (event: MouseEvent) => void;
  /**
   * Color profile to use with this ButtonCircle.
   */
  color?: 'join' | 'cancel' | 'message';
  /**
   * Whether or not this ButtonCircle has a transparent background.
   */
  ghost?: boolean;

  /**
   * Whether or not this ButtonCircle has an outline/border.
   */
  outline?: boolean;

  /**
   * Child components of this ButtonCircle.
   *
   * @deprecated Use `prefix-icon` prop instead.
   */
  children?: ReactNode;

  shallowDisabled?: boolean;

  /**
   * Whether or not to stop the event from bubbling up.
   */
  stopPropagation?: boolean;
};
