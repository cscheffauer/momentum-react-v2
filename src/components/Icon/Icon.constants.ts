import { IconScale } from '.';
import type { IconWeight } from './Icon.types';

const CLASS_PREFIX = 'md-icon';
const COLOR_INHERIT = 'currentColor';

const WEIGHTS: Record<string, IconWeight> = {
  light: 'light',
  regular: 'regular',
  bold: 'bold',
  filled: 'filled',
};

const SCALES = {
  25: 25,
  50: 50,
  75: 75,
  100: 100,
  125: 125,
  150: 150,
  175: 175,
  200: 200,
};

const SIZES: Record<IconScale, IconScale> = {
  8: 8,
  10: 10,
  12: 12,
  14: 14,
  16: 16,
  18: 18,
  20: 20,
  22: 22,
  24: 24,
  28: 28,
  32: 32,
  36: 36,
  40: 40,
  48: 48,
  56: 56,
  64: 64,
  120: 120,
  124: 124,
  auto: 'auto',
  inherit: 'inherit',
};

const DEFAULTS = {
  WEIGHT: WEIGHTS.regular,
  AUTO_SCALE: false,
  WEIGHTLESS: false,
};

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
  notFound: `${CLASS_PREFIX}-not-found`,
  coloured: `${CLASS_PREFIX}-coloured`,
  noShrink: `${CLASS_PREFIX}-no-shrink`,
  autoScales: `${CLASS_PREFIX}-auto-scales`,
  scales: `${CLASS_PREFIX}-scales`,
};

export { CLASS_PREFIX, COLOR_INHERIT, DEFAULTS, SCALES, SIZES, STYLE, WEIGHTS };
