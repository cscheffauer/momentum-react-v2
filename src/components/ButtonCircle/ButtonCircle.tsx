/* eslint-disable @typescript-eslint/ban-ts-comment */
import React, { Children, forwardRef, ReactElement, RefObject } from 'react';
import classnames from 'classnames';
import { Button as MdcButton } from '@momentum-design/components/dist/react';
import type { Button, ButtonVariant } from '@momentum-design/components';
import type IconKeys from '@momentum-design/icons/dist/types/types';

import { Props } from './ButtonCircle.types';
import './ButtonCircle.style.scss';
import { DEFAULTS, STYLE } from './ButtonCircle.constants';

const ButtonCircle = forwardRef((props: Props, providedRef: RefObject<Button>) => {
  const {
    color,
    className,
    size = DEFAULTS.SIZE,
    children,
    onPress,
    ghost,
    outline,
    shallowDisabled,
    stopPropagation = DEFAULTS.STOP_PROPAGATION,
    ...rest
  } = props;

  let newColor;
  switch (color) {
    case 'join':
      newColor = 'positive';
      break;
    case 'cancel':
      newColor = 'negative';
      break;
    case 'message':
      newColor = 'accent';
      break;
    default:
      newColor = 'default';
      break;
  }

  let variant: ButtonVariant = 'primary';

  if (ghost) {
    variant = 'tertiary';
  }
  if (outline) {
    variant = 'secondary';
  }

  let prefixIcon;
  if (children && Children.count(children) === 1) {
    const Icon = children as ReactElement;
    // @ts-ignore
    if (Icon.type?.displayName === 'Mrv2Icon') {
      prefixIcon = [Icon.props.name, Icon.props.weight || 'regular'].join('-');
    }
  }

  // this is a workaround for now for cases where usePress from react-aria is used on a parent
  // of the button. We want to avoid the event bubbling up to the parent in those cases
  // this can be removed once ListItemBase is refactored to use the new momentum-design listitem component
  const preventBubble = (event: any) => {
    // Prevent the event from bubbling up to the parent element
    if (stopPropagation) {
      event.stopPropagation();
      event.preventDefault();
    }
  };

  const handleClick = (event: MouseEvent) => {
    preventBubble(event);
    if (onPress) {
      onPress(event);
    }
  };

  return (
    <MdcButton
      prefixIcon={prefixIcon as IconKeys}
      variant={variant}
      color={newColor}
      ref={providedRef}
      size={size}
      onClick={handleClick}
      onPointerUp={preventBubble}
      onPointerDown={preventBubble}
      softDisabled={shallowDisabled}
      className={classnames({ [STYLE.widthOverride]: !!prefixIcon }, STYLE.wrapper, className)}
      {...rest}
    >
      {!prefixIcon && children}
    </MdcButton>
  );
});

ButtonCircle.displayName = 'ButtonCircle';

export default ButtonCircle;
