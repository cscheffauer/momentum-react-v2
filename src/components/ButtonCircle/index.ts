import type { IconButtonSize } from '@momentum-design/components';
import { default as ButtonCircle } from './ButtonCircle';
import type { Props } from './ButtonCircle.types';

export type ButtonCircleProps = Props;
export type ButtonCircleSize = IconButtonSize;

export default ButtonCircle;
