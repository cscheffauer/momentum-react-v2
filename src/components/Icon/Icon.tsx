import React from 'react';
import './Icon.style.scss';
import { Props } from './Icon.types';
import { DEFAULTS, STYLE } from './Icon.constants';
import classnames from 'classnames';
import { getResolvedSVGName } from './Icon.utils';
import Tooltip from '../Tooltip';
import { Icon as MdcIcon } from '@momentum-design/components/dist/react';
import type IconKeys from '@momentum-design/icons/dist/types/types';

/**
 * Icon component that can dynamically display SVG icons with a valid name.
 */
const Icon: React.FC<Props> = (props: Props) => {
  const {
    autoScale,
    className,
    id,
    name,
    scale,
    style,
    title,
    weight,
    weightless = DEFAULTS.WEIGHTLESS,
    ariaLabel,
    tooltipProps,
    fillColor,
    ...otherProps
  } = props;
  const resolvedSVGName = getResolvedSVGName(name, weight, weightless);

  const accessibleName = ariaLabel || otherProps['aria-label'] || title;

  delete otherProps['aria-label'];

  const isTooltipWrapped = tooltipProps?.children;

  const fillColorStyle = fillColor ? { '--mdc-icon-fill-color': fillColor } : {};

  const content = (
    <MdcIcon
      className={classnames(
        STYLE.wrapper,
        STYLE.autoScales,
        STYLE.scales,
        STYLE.noShrink,
        className
      )}
      data-test={name}
      data-scale={!autoScale && scale}
      data-autoscale={autoScale || DEFAULTS.AUTO_SCALE}
      name={resolvedSVGName as IconKeys}
      aria-label={isTooltipWrapped ? undefined : accessibleName}
      tabIndex={isTooltipWrapped ? 0 : undefined}
      style={{ ...fillColorStyle, ...style }}
      id={id}
      {...otherProps}
    />
  );

  return isTooltipWrapped ? (
    <Tooltip
      placement="top"
      strategy="fixed"
      type="label"
      triggerComponent={content}
      {...tooltipProps}
    />
  ) : (
    content
  );
};

Icon.displayName = 'Mrv2Icon';

export default Icon;
