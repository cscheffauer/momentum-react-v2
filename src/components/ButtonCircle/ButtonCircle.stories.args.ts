import {
  commonMdcButton,
  commonHTMLAttributes,
  commonStyles,
} from '../../storybook/helper.stories.argtypes';

import { BUTTON_VARIANTS, ICON_BUTTON_SIZES, BUTTON_COLORS } from '@momentum-design/components';

// buttoncircle does not support children
delete commonMdcButton.children;

const buttonCircleArgTypes = {
  prefixIcon: {
    description: 'The icon to display in the button.',
    control: { type: 'text' },
    table: {
      type: {
        summary: 'string',
      },
      defaultValue: {
        summary: undefined,
      },
    },
  },
  variant: {
    description: 'Modifies the variant of this component.',
    options: [undefined, ...Object.values(BUTTON_VARIANTS)],
    control: { type: 'select' },
    table: {
      type: {
        summary: 'string',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
  size: {
    description:
      'Modifies the size of this component. \n\n**NOTE:** Size 20 is only supported for `variant="tertiary"`',
    options: [undefined, ...Object.values(ICON_BUTTON_SIZES)],
    control: { type: 'select' },
    table: {
      type: {
        summary: 'number',
      },
      defaultValue: {
        summary: 32,
      },
    },
  },
  color: {
    description:
      'Modifies the color of this component. \n\n**NOTE:** Coloring is only supported for variants `primary` & `secondary`',
    options: [undefined, ...Object.values(BUTTON_COLORS)],
    control: { type: 'select' },
    table: {
      type: {
        summary: 'string',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
  inverted: {
    description: 'Inverts the color of the button.',
    control: { type: 'boolean' },
    table: {
      type: {
        summary: 'boolean',
      },
      defaultValue: {
        summary: false,
      },
    },
  },
};

export { buttonCircleArgTypes };

export default {
  ...commonStyles,
  ...commonMdcButton,
  ...commonHTMLAttributes,
  ...buttonCircleArgTypes,
};
