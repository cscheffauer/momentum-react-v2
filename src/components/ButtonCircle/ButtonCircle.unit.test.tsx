import React from 'react';
import { screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';

import ButtonCircle from './';
import { renderWithWebComponent } from '../../../test/utils';
import Icon from '../Icon';

describe('<ButtonCircle />', () => {
  const setup = async (component: React.ReactElement) => {
    return renderWithWebComponent(component);
  };

  describe('snapshot', () => {
    it('should match snapshot', async () => {
      const { container } = await setup(<ButtonCircle />);
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with multiple children', async () => {
      const { container } = await setup(
        <ButtonCircle>
          <div>X</div>
          <div>Y</div>
        </ButtonCircle>
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with className', async () => {
      const className = 'example-class';
      const { container } = await setup(<ButtonCircle className={className} />);
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with id', async () => {
      const id = 'example-id';
      const { container } = await setup(<ButtonCircle id={id} />);
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with style', async () => {
      const style = { color: 'pink' };
      const { container } = await setup(<ButtonCircle style={style} />);
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with color', async () => {
      const color = 'join';
      const { container } = await setup(<ButtonCircle color={color}>X</ButtonCircle>);
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with size', async () => {
      const size = 52;
      const { container } = await setup(<ButtonCircle size={size}>X</ButtonCircle>);
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot when disabled', async () => {
      const { container } = await setup(<ButtonCircle disabled>X</ButtonCircle>);
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot when shallow disabled', async () => {
      const { container } = await setup(<ButtonCircle shallowDisabled>X</ButtonCircle>);
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot when a ghost', async () => {
      const { container } = await setup(<ButtonCircle ghost>X</ButtonCircle>);
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot when color is outlined', async () => {
      const { container } = await setup(<ButtonCircle outline>X</ButtonCircle>);
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with title', async () => {
      const title = 'Example Text';
      const { container } = await setup(<ButtonCircle title={title} />);
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot when color is outlined and inverted is set', async () => {
      const { container } = await setup(<ButtonCircle outline inverted />);
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot when prefixIcon is set', async () => {
      const { container } = await setup(<ButtonCircle prefixIcon="placeholder-regular" />);
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot when variant is set', async () => {
      const { container } = await setup(<ButtonCircle variant="secondary" />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have variant set to secondary when outline is provided', async () => {
      await setup(<ButtonCircle outline />);
      expect(screen.getByRole('button')).toHaveAttribute('variant', 'secondary');
    });

    it('should have variant set to tertiary when ghost is provided', async () => {
      await setup(<ButtonCircle ghost />);
      expect(screen.getByRole('button')).toHaveAttribute('variant', 'tertiary');
    });

    it('should have variant set to primary when no variant is provided', async () => {
      await setup(<ButtonCircle />);
      expect(screen.getByRole('button')).toHaveAttribute('variant', 'primary');
    });

    it('should have color set to positive when color is join', async () => {
      await setup(<ButtonCircle color="join" />);
      expect(screen.getByRole('button')).toHaveAttribute('color', 'positive');
    });

    it('should have color set to negative when color is cancel', async () => {
      await setup(<ButtonCircle color="cancel" />);
      expect(screen.getByRole('button')).toHaveAttribute('color', 'negative');
    });

    it('should have color set to accent when color is message', async () => {
      await setup(<ButtonCircle color="message" />);
      expect(screen.getByRole('button')).toHaveAttribute('color', 'accent');
    });

    it('should have color set to default when no color is provided', async () => {
      await setup(<ButtonCircle />);
      expect(screen.getByRole('button')).toHaveAttribute('color', 'default');
    });

    it('should have prefixIcon not set when no Icon child is provided', async () => {
      await setup(
        <ButtonCircle>
          <div>Not an Icon</div>
          <div>Also not an Icon</div>
        </ButtonCircle>
      );
      expect(screen.getByRole('button')).not.toHaveAttribute('prefix-icon');
    });

    it('should have prefixIcon set to the icon name when a single Icon child is provided', async () => {
      await setup(
        <ButtonCircle>
          <Icon name="placeholder" weight="bold" />
        </ButtonCircle>
      );
      expect(screen.getByRole('button')).toHaveAttribute('prefix-icon', 'placeholder-bold');
    });

    it('should have prefixIcon set to the icon name and regular weight when a single Icon child without weight is provided', async () => {
      await setup(
        <ButtonCircle>
          <Icon name="placeholder" />
        </ButtonCircle>
      );
      expect(screen.getByRole('button')).toHaveAttribute('prefix-icon', 'placeholder-regular');
    });
  });

  describe('actions', () => {
    it('should call onPress when clicked', async () => {
      const onPress = jest.fn();
      await setup(<ButtonCircle onPress={onPress} />);
      await userEvent.click(screen.getByRole('button'));
      expect(onPress).toHaveBeenCalled();
    });

    it('should not call onPress when disabled', async () => {
      const onPress = jest.fn();
      await setup(<ButtonCircle onPress={onPress} disabled />);
      await userEvent.click(screen.getByRole('button'));
      expect(onPress).not.toHaveBeenCalled();
    });

    it('should call onPress when shallowDisabled', async () => {
      const onPress = jest.fn();
      await setup(<ButtonCircle onPress={onPress} shallowDisabled />);
      await userEvent.click(screen.getByRole('button'));
      expect(onPress).toHaveBeenCalled();
    });
  });
});
