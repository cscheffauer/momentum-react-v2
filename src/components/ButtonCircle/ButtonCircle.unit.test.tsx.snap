// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonCircle /> snapshot should match snapshot 1`] = `
<div>
  <mdc-button
    class="md-button-circle-wrapper"
    color="default"
    role="button"
    size="40"
    tabindex="0"
    type="button"
    variant="primary"
  />
</div>
`;

exports[`<ButtonCircle /> snapshot should match snapshot when a ghost 1`] = `
<div>
  <mdc-button
    class="md-button-circle-wrapper"
    color="default"
    data-btn-type="pill"
    role="button"
    size="40"
    tabindex="0"
    type="button"
    variant="tertiary"
  >
    X
  </mdc-button>
</div>
`;

exports[`<ButtonCircle /> snapshot should match snapshot when color is outlined 1`] = `
<div>
  <mdc-button
    class="md-button-circle-wrapper"
    color="default"
    data-btn-type="pill"
    role="button"
    size="40"
    tabindex="0"
    type="button"
    variant="secondary"
  >
    X
  </mdc-button>
</div>
`;

exports[`<ButtonCircle /> snapshot should match snapshot when color is outlined and inverted is set 1`] = `
<div>
  <mdc-button
    class="md-button-circle-wrapper"
    color="default"
    inverted=""
    role="button"
    size="40"
    tabindex="0"
    type="button"
    variant="secondary"
  />
</div>
`;

exports[`<ButtonCircle /> snapshot should match snapshot when disabled 1`] = `
<div>
  <mdc-button
    aria-disabled="true"
    class="md-button-circle-wrapper"
    color="default"
    data-btn-type="pill"
    disabled=""
    internals-disabled=""
    role="button"
    size="40"
    tabindex="-1"
    type="button"
    variant="primary"
  >
    X
  </mdc-button>
</div>
`;

exports[`<ButtonCircle /> snapshot should match snapshot when prefixIcon is set 1`] = `
<div>
  <mdc-button
    class="md-button-circle-wrapper"
    color="default"
    data-btn-type="icon"
    prefix-icon="placeholder-regular"
    role="button"
    size="40"
    tabindex="0"
    type="button"
    variant="primary"
  />
</div>
`;

exports[`<ButtonCircle /> snapshot should match snapshot when shallow disabled 1`] = `
<div>
  <mdc-button
    aria-disabled="true"
    class="md-button-circle-wrapper"
    color="default"
    data-btn-type="pill"
    role="button"
    size="40"
    soft-disabled=""
    tabindex="0"
    type="button"
    variant="primary"
  >
    X
  </mdc-button>
</div>
`;

exports[`<ButtonCircle /> snapshot should match snapshot when variant is set 1`] = `
<div>
  <mdc-button
    class="md-button-circle-wrapper"
    color="default"
    role="button"
    size="40"
    tabindex="0"
    type="button"
    variant="secondary"
  />
</div>
`;

exports[`<ButtonCircle /> snapshot should match snapshot with className 1`] = `
<div>
  <mdc-button
    class="md-button-circle-wrapper example-class"
    color="default"
    role="button"
    size="40"
    tabindex="0"
    type="button"
    variant="primary"
  />
</div>
`;

exports[`<ButtonCircle /> snapshot should match snapshot with color 1`] = `
<div>
  <mdc-button
    class="md-button-circle-wrapper"
    color="positive"
    data-btn-type="pill"
    role="button"
    size="40"
    tabindex="0"
    type="button"
    variant="primary"
  >
    X
  </mdc-button>
</div>
`;

exports[`<ButtonCircle /> snapshot should match snapshot with id 1`] = `
<div>
  <mdc-button
    class="md-button-circle-wrapper"
    color="default"
    id="example-id"
    role="button"
    size="40"
    tabindex="0"
    type="button"
    variant="primary"
  />
</div>
`;

exports[`<ButtonCircle /> snapshot should match snapshot with multiple children 1`] = `
<div>
  <mdc-button
    class="md-button-circle-wrapper"
    color="default"
    data-btn-type="pill"
    role="button"
    size="40"
    tabindex="0"
    type="button"
    variant="primary"
  >
    <div>
      X
    </div>
    <div>
      Y
    </div>
  </mdc-button>
</div>
`;

exports[`<ButtonCircle /> snapshot should match snapshot with size 1`] = `
<div>
  <mdc-button
    class="md-button-circle-wrapper"
    color="default"
    data-btn-type="pill"
    role="button"
    size="32"
    tabindex="0"
    type="button"
    variant="primary"
  >
    X
  </mdc-button>
</div>
`;

exports[`<ButtonCircle /> snapshot should match snapshot with style 1`] = `
<div>
  <mdc-button
    class="md-button-circle-wrapper"
    color="default"
    role="button"
    size="40"
    style="color: pink;"
    tabindex="0"
    type="button"
    variant="primary"
  />
</div>
`;

exports[`<ButtonCircle /> snapshot should match snapshot with title 1`] = `
<div>
  <mdc-button
    class="md-button-circle-wrapper"
    color="default"
    role="button"
    size="40"
    tabindex="0"
    title="Example Text"
    type="button"
    variant="primary"
  />
</div>
`;
